collect_app/src/main/java/org/odk/collect/android/tasks/InstanceUploaderTask.java

```diff
@@ -108,7 +109,7 @@ public Outcome doInBackground(Long... instanceIdsToUpload) {
             publishProgress(i + 1, instancesToUpload.size());
 
             if (completeDestinationUrl != null) {
-                Analytics.log(AnalyticsEvents.INSTANCE_UPLOAD_CUSTOM_SERVER);
+                Analytics.log(AnalyticsEvents.INSTANCE_UPLOAD_CUSTOM_SERVER, "referrer", referrer);
```

Comment:
@lognaturel logging like this limits us to viewing a max of 200 distinct values right? As far as I'm aware there isn't another way to log custom data though (other than a user property which isn't really appropriate).

Also assuming we do want to do an event. It needs to be setup on Firebase as far as I can remember.

Reply:
The 200 limit is right and I'll be absolutely flabbergasted if there are more than 5-10 uniques!!

> It needs to be setup on Firebase as far as I can remember.

Ooh, good call. I don't remember that part at all. Would you be ok doing it when you have some time (no big rush)?